module/apmawssdkgo/session_test.go

```diff
@@ -0,0 +1,77 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// +build go1.13
+
+package apmawssdkgo // import "go.elastic.co/apm/module/apmawssdkgo"
+
+import (
+	"bytes"
+	"context"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"go.elastic.co/apm/apmtest"
+)
+
+func TestS3(t *testing.T) {
+	region := "us-west-2"
+	cfg := aws.NewConfig().
+		WithRegion(region).
+		WithDisableSSL(true).
+		WithCredentials(credentials.AnonymousCredentials)
+
+	session := session.Must(session.NewSession(cfg))
+	spanSubtype := "s3"
+	spanType := serviceTypeMap[spanSubtype]
+
+	bucketName := "BUCKET"
+	uploader := s3manager.NewUploader(WrapSession(session))
+	tx, spans, errors := apmtest.WithTransaction(func(ctx context.Context) {
+		uploader.UploadWithContext(ctx, &s3manager.UploadInput{
+			Bucket: aws.String(bucketName),
+			Key:    aws.String("some key"),
+			Body:   bytes.NewBuffer([]byte("some random body")),
+		})
+	})
+
+	require.Len(t, spans, 1)
+	require.Len(t, errors, 1)
```

Comment:
should we check the error contents too?

Reply:
sure. While I was adding some checks, I read that error.ParentID is the transaction id:

https://github.com/elastic/apm-agent-go/blob/master/model/model.go#L458-L460

But from the test, error.ParentID is actually the span.ID:

```go
assert.Equal(t, tx.ID, err.TransactionID)
assert.Equal(t, span.ID, err.ParentID)
```

Is the code correct but the comment is wrong?

Reply:
oops, yeah sorry - the comment is wrong. It should be "span or transaction"